### PR TITLE
Add JSInspect to CI

### PR DESCRIPTION
**Pull request by montymxb**
Proposes to add [jsinspect](https://github.com/danielstjules/jsinspect) to our CI in order to catch and flag code duplication issues. As is there are several files that contain nearly identical logic, which makes maintenance/updates not quite as easy (or error free) as it should be. Ideally this should help force us to consolidate logic rather than to duplicate it, whether by intention or accident.

This is in response to recent issues #3432 and #3451. The overall checking process is rather quick, and shouldn't cause any noticeable issues with overhead or increased time for tests.

### CODE REVIEW

**Comment by montymxb** [comment]
Tests are expected to fail as is due to present duplications in the codebase. Assuming this is a feature that we would be interested in I would provide the necessary changes to pass in the same PR here.

File: .travis.yml [hunk 1]
```diff
@@ -35,11 +35,14 @@ before_install:
...
 - nvm use $NODE_VERSION
 before_script:
+- npm install -g jsinspect
 - node -e 'require("./lib/index.js")'
 - psql -c 'create database parse_server_postgres_adapter_test_database;' -U postgres
 - psql -c 'CREATE EXTENSION postgis;' -U postgres -d parse_server_postgres_adapter_test_database
 - psql -c 'CREATE EXTENSION postgis_topology;' -U postgres -d parse_server_postgres_adapter_test_database
 - silent=1 mongodb-runner --start
+script:
+- jsinspect ./src && npm test
```
Comment (flovilmart) [inline .travis.yml]: we could run in before_script right? any failure would be caught.

Comment (montymxb) [reply .travis.yml]: Yeah we can do that if we don't want to fudge with just `script`.

**Comment by flovilmart** [comment]
@montymxb after all those merges, it's kinda borked now :/

**Comment by montymxb** [comment]
Yeah just a bit 🤣 . I'm thinking it would be best to close this as nonessential. Instead I'll cherry pick some changes and present them without forcing jsinspect. There were quite a few that would make good additions, but I think they would be more appropriate as separate PRs. Not only that but it would be easier to review if it were small groups of related changes, one at a time.

**Pull request closed without merging**